cpp/src/DataStorm/Instance.cpp

```diff
@@ -103,7 +102,7 @@ Instance::Instance(const shared_ptr<Ice::Communicator>& communicator) : _communi
 
     _executor = make_shared<CallbackExecutor>();
     _connectionManager = make_shared<ConnectionManager>(_executor);
-    _timer = make_shared<Timer>();
+    _timer = make_shared<Ice::Timer>();
```

Comment:
Switch to use the Ice::Timer class and got rid of the custom DataStorm::Timer class, this is causing problems:

- The code schedules tasks after it has been destroyed, which is not allowed by Ice::Timer, but was allowed by DataStorm::Timer.
- I have to wrap the closures we pass to schedule, as Ice::Timer only schedules TimerTask instances, see https://github.com/zeroc-ice/ice/issues/2877